app/javascript/lib/action.js

```diff
@@ -64,7 +66,7 @@ export default {
   initialize: makeAction("initialize", ["players", "soloMode", "variant", "baseGame"]),
   blockCanal: makeAction("blockCanal", []),
   bondPurchase: makeAction("bondPurchase", ["nation", "player", "cost", "tradeInValue"]),
-  buildFactory: makeAction("buildFactory", ["province"]),
+  buildFactory: makeAction("buildFactory", ["nation", "province", "player", "nationCosts", "playerCosts"]),
```

Comment:
Do we need to specify `nation` here? Aren't all provinces hardcoded to which nation they belong (if any)?

Reply:
Ah yeah you're right, we can get `nation` through `graph.get(province).nation`, can't we?
Will fix

Reply:
Actually, I'm blind

Fixed here [06a08f6](https://github.com/Thrillberg/imperial/pull/711/commits/06a08f6ea787c95d59de0cb126cfac2ce7479ea4)